Classes/StenoSignal.sc

```diff
@@ -53,43 +69,79 @@ StenoSignal {
-		DetectSilence.ar(detectSignal, time: 0.01, doneAction:2); // free the synth when gate = 0 and fx output is silent
-		signal = XFade2.ar(drySignal, signal, mix * 2 - 1); // mix in filter output to dry signal.
-		signal = signal + (oldSignal * max(through, 1 - env)); // when the gate is switched off (released), let old input through
-
-		// remove hanging notes if necessary:
+
+		// gating analysis
 		gateHappened = gate <= 0;
-		FreeSelf.kr(TDelay.kr(gateHappened, max(fadeTime, \hangTime.kr(30))) + (gateHappened * \steno_unhang.tr(0)));
+		dcBlocked = LeakDC.ar(signal.sum);
+
+		// free synth if signal constant for fadeTime:
+		DetectSilence.ar(max(gate, dcBlocked), time: fadeTime, doneAction:2);
+
+		// if signal not constant, remove hanging notes some time after release
+		FreeSelf.kr(
+			TDelay.kr(gateHappened, max(fadeTime, \hangTime.kr(30)))      //  or
+			// + (gateHappened * \steno_unhang.tr(0))
+		);
+
+
+		oldSignal = In.ar(outBus + offset, argNumChannels); // previous signal on bus
+		drySignal = In.ar(dryIn  + offset, argNumChannels); // dry signal (mostly same as oldSignal but may come from another bus)
+		tailSignal = In.ar(tailBus + offset, argNumChannels); // tails from replaced synths
+		
+		signal = Mix.ar([
+			  // mix filter output with dry signal if synth is not released
+			  // this part is supposed to fade out itself (envelope assigned at filter input)
+			signal,
+			  // collect tails
+			tailSignal, 
+			
+			  // fade old signal according to 1-env 
+			  // `through` is used to carry original signal in the parallel case
+			oldSignal * max(through, 1 - env)
+		]);
+		// assign mix if synth is not released
+		signal = XFade2.ar(drySignal, signal, MulAdd(max(mix, 1-gate), 2, -1));
+
 		this.addOutput(signal, offset);
 	}
 
 	// set quelle output
 	quelleOutput { |signal, argNumChannels, offset = 0|
-		var localMix, localInput, oldSignal;
+		var oldSignal, tailSignal;
 		argNumChannels = min(argNumChannels ? numChannels, numChannels - offset); // avoid overrun of total channels given
 
 		signal = signal.asArray.keep(argNumChannels);
 		if(multiChannelExpand) { signal = signal.wrapExtend(argNumChannels) };
 
-		oldSignal = In.ar(outBus + offset, argNumChannels);          // previous signal on bus
-		localInput = this.quelleInput(argNumChannels, offset);
+		oldSignal  = In.ar(outBus + offset, argNumChannels);  // previous signal on bus
+		tailSignal = In.ar(tailBus + offset, argNumChannels); // tails from replaced synths
+
+
+		signal = Mix.ar([
+			signal * env,
 
-		signal = XFade2.ar(oldSignal, signal + oldSignal, (mix * env) * 2 - 1);  // can't use Out here, because "in" can be different than "out"
-		FreeSelfWhenDone.kr(env);
+			  // collect tails
+			tailSignal,
+
+			  // only add old signal if last in replaceGroup
+			  // TODO: if release (not replace), continue mixing
+			oldSignal * gate
+		]);
+		// assign mix if synth is not released
+		signal = signal * max(mix, 1-gate);
```

Comment:
it would be interesting if we could factor out the `mix` parameter into `writeToBus` ? Looking from far away, `mix` is really quite elementary and perhaps should not be up to the signal handling. 

Just an idea.